### PR TITLE
Refactor and improvements for "transactional-updates" module

### PR DESCRIPTION
**Pull request by meaksh**
### What does this PR do?

This PR changes the way the `transactional-update` module is working internally in order to run Salt actions inside new transactions.

The previous architecture used the Salt SSH approach, using the SSH wrappers, Salt thin and `state.pkg` in order to execute a Salt action inside a transaction. We found this is a bit problematic, and produce some issues particularly when dealing with complex/fancy states like:

```
# this is example1.sls

my_test_state:
  module.run:
    - name: state.apply
```

The above state produce the following error when running in a transactional system:

```
slem51:~ # salt-call state.apply example1
local:
----------
          ID: test_example1
    Function: module.run
        Name: state.apply
      Result: False
     Comment: Module function state.apply executed
     Started: 10:42:19.295456
    Duration: 71.843 ms
     Changes:   
              ----------
              ret:
                  ----------
                  no_|-states_|-states_|-None:
                      ----------
                      __run_num__:
                          0
                      changes:
                          ----------
                      comment:
                          No Top file or master_tops data matches found. Please see master log for details.
                      name:
                          No States
                      result:
                          False

Summary for local
------------
Succeeded: 0 (changed=1)
Failed:    1
------------
Total states run:     1
Total run time:  71.843 ms

```

Similarly with this other state, where filerefs appears as missing (will force us to use extra_filerefs for this particular call):

```
# this is example2.sls

run_test2:
  module.run:
    - name: state.apply
    - mods: tests2
```

and 

```
# this is tests2.sls

something_test:
  cmd.run:
    - name: date
```

In this case we are getting:

```
slem51:~ # salt-call state.apply example2
local:
----------
          ID: run_test2
    Function: module.run
        Name: state.apply
      Result: True
     Comment: Module function state.apply executed
     Started: 10:45:22.012088
    Duration: 89.861 ms
     Changes:   
              ----------
              ret:
                  - No matching sls found for 'tests2' in env 'base'

Summary for local
------------
Succeeded: 1 (changed=1)
Failed:    0
------------
Total states run:     1
```

What this PR does in order to solve this problems is the following:

1) Use a normal "salt-call" execution instead SSH approach to execute a Salt action inside a new transaction.
2) Make use of `concurrent` flag to allow execution of the underlying "salt-call" in case of states execution.
3) Implements `--no-return-event` for "salt-call" to avoid getting duplicated events on the Salt event bus.
4) Solved the above issues about missing filerefs or master top data.
5) Fixed a problem with `state.highstate` not acting according to `concurrent` flag described in Salt documentation.

This brings some benefits:

- No need to struggle with "extra_filerefs" for those states that contains references that are not included in the state.pkg.
- More transparent approach. Should cause less troubles and work with complex states.

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
- [x] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with GPG?
Yes

Please review [Salt's Contributing Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

File: salt/modules/transactional_update.py [hunk 1]
```diff
@@ -950,65 +941,48 @@ def call(function, *args, **kwargs):
...
                 "--",
                 function,
             ]
             + list(args)
             + ["{}={}".format(k, v) for (k, v) in safe_kwargs.items()]
         )
+
         try:
             ret_stdout = run([str(x) for x in salt_argv], snapshot="continue")
+            # FIXME: This is hack to workaround tukit issue. It just removed the conflicting line
```
Comment (aplanas) [inline salt/modules/transactional_update.py:963]: This was fixed in tukit since 3.5.7 IIRC.

**Comment by meaksh** [comment]
I notice there are few tests failures, I'm pushing necessary fixes

File: salt/modules/transactional_update.py (hunk 1 above)
Comment (meaksh) [reply salt/modules/transactional_update.py:963]: Agreed to remove this from here :+1:

**Review by twangboy — DISMISSED** [review]

**Review by Ch3LL — CHANGES_REQUESTED** [review]

File: salt/modules/transactional_update.py [hunk 2]
```diff
@@ -1044,49 +1012,7 @@ def apply_(mods=None, **kwargs):
...
-def sls(
-    mods,
-    saltenv="base",
-    test=None,
-    exclude=None,
-    activate_transaction=False,
-    queue=False,
-    **kwargs
-):
+def sls(mods, activate_transaction=False, queue=False, **kwargs):
```
[the remark this answers is not in the document]
Comment (meaksh) [reply salt/modules/transactional_update.py:1015]: Well, I'm not actually deprecating any parameter here. The function keeps accepting the same, it is just that we pass them now as part of `kwargs`.

This `transactional_update.sls` function is just a wrapper on top of `state.sls`, passing all passed arguments as kwargs. It allows same parameters as `state.sls`.

File: salt/modules/transactional_update.py [hunk 3]
```diff
@@ -1216,40 +1101,15 @@ def highstate(activate_transaction=False, queue=False, **kwargs):
...
+    return call(
+        "state.highstate",
+        activate_transaction=activate_transaction,
+        concurrent=True,
+        **kwargs
     )
 
 
-def single(fun, name, test=None, activate_transaction=False, queue=False, **kwargs):
+def single(fun, name, activate_transaction=False, queue=False, **kwargs):
```
[the remark this answers is not in the document]
Comment (meaksh) [reply salt/modules/transactional_update.py:1112]: Same reasoning as above here. No deprecation but passing parameters in kwargs.

**Comment by meaksh** [comment]
@Ch3LL thanks for the review! :+1:

I've pushed some of the requested changes and did some comments about the parameter deprecations.

**Review by Ch3LL — APPROVED** [review]

**Review by garethgreenaway — APPROVED** [review]

**Pull request merged**